docs/DeveloperGuide.md

```diff
@@ -319,12 +317,93 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
   Use case ends.
 
-* 3a. The given index is invalid.
+**Use case: UC07 Filter by Tag**
 
-    * 3a1. AddressBook shows an error message.
+**MSS**
+
+1. User filters for contacts with a specified tag.
+2. WedLinker only shows a list of contact with the specified tag.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+**Use case: UC08 Create Tags**
+
+**MSS**
+
+1. User creates a tag.
+2. WedLinker displays the successful creation of tag.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The tag already exists.
+* 2b. WedLinker does not create a new tag.
+* 2c. WedLinker informs the user the tag already exists.
+
+  Use case ends.
+
+**Use case: UC09 Tagging a contact with a specified tag**
+
+**MSS**
+
+1. User searches for the contact to be tagged (UC05).
+2. WedLinker shows a list of contacts for the search.
+3. User looks for the index to be tagged.
+4. User adds the tag to the contact.
+5. WedLinker informs the user the contact is tagged.
+6. WedLinker shows the user the final result of the contact.
+
+   Use case ends.
+
+**Extensions**
+
+* 2a. The list is empty.
+
+  Use case ends.
+
+* 4a. WedLinker detects that the tag does not exist.
+
+    * 4a1. WedLinker informs the user the tag failed.
+
+      Use case ends.
+
+* 4b. The given index is invalid.
+
+    * 4b1. WedLinker shows an error message.
```

Comment:
I think there is a bit of inconsistency with Dasha's use cases - for hers she wrote "shows an error message prompting the user to input an invalid index", perhaps can standardise? Personally i prefer having a bit of elaboration on what the error message is about

Reply:
I will make the changes.